kairos-contracts/demo-contract/contract/src/main.rs

```diff
@@ -81,19 +88,69 @@ pub extern "C" fn deposit() {
     casper_event_standard::emit(new_deposit_record);
 }
 
+#[derive(serde::Serialize, serde::Deserialize)]
+pub struct Proof {
+    pub receipt: risc0_zkvm::Receipt,
+    pub program_id: [u32; 8],
+}
```

Comment:
The contract must have it's program id compiled in as a constant or to enable upgradability gotten from a uref owned by the l2 operator or DAO.

Reply:
We can add this as a post-demo feature. If you insist that we do it prior to the demo I can just hardcode it for the circuit right before we demo.

Reply:
This has been resolved, we will be passing a serialized receipt.